### PR TITLE
feat: Add an Elm module

### LINKED ISSUE #873: Add Elm module
#### Is your feature request related to a problem? Please describe.
Currently, there is no support for [elm](https://elm-lang.org/).

#### Describe the solution you'd like
An implementation similar to [the one](https://github.com/denysdovhan/spaceship-prompt/pull/528) in [spaceship-prompt](https://github.com/denysdovhan/spaceship-prompt) would be nice and shouldn't be too difficult to support.

### PR DESCRIPTION
**Pull request by m0nhawk**
#### Description

* Add an Elm module if it detects one of `elm.json`, `elm-package.json`, `elm-stuff` directory or `*.elm` files.

#### Motivation and Context

Closes #873

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Screenshots (if appropriate):
Screen Shot 2020-01-25 at 2 04 06 PM

#### How Has This Been Tested?

- [x] I have tested using **MacOS**
- [x] I have tested using **Linux**
- [x] I have tested using **Windows**

#### Checklist:

- [x] I have updated the documentation accordingly.
- [x] I have updated the tests accordingly.

### CODE REVIEW

**Comment by m0nhawk** [comment]
I have tested this on macOS (screenshot), Linux (latest Fedora) and Windows.

Unfortunately, the `ghaction-chocolatey` doesn't export variables nicely (I believe it's similar I had with Haskell Support in #546) and it fails to run `elm` on Windows in Github Actions.

**Comment by m0nhawk** [comment]
@andytom Thank you. Somehow I missed few lines to fix.

File: docs/config/README.md [hunk 1]
```diff
@@ -95,6 +95,7 @@ prompt_order = [
     "hostname",
     "kubernetes",
     "directory",
+    "elm",
```
Comment (jonstodle) [inline docs/config/README.md]: This should be moved down, together with the other languages/frameworks

File: src/modules/elm.rs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        .set_extensions(&["elm"])
+        .set_folders(&["elm-stuff"])
+        .is_match();
+
+    if !is_elm_project {
+        return None;
+    }
+
+    let elm_version = utils::exec_cmd("elm", &["--version"])?.stdout;
+    let formatted_version = format_elm_version(&elm_version)?;
```
Comment (jonstodle) [inline src/modules/elm.rs]: Is there a reason for moving `format!("v{}", elm_version.trim())` to a seperate function?

File: docs/config/README.md (hunk 1 above)
Comment (m0nhawk) [reply docs/config/README.md]: Fixed. 👍

File: src/modules/elm.rs (hunk 2 above)
Comment (m0nhawk) [reply src/modules/elm.rs]: I agree. I addressed this.

**Comment by m0nhawk** [comment]
@matchai I made the necessary changes to implement new tests (at least the tests are passing for this module).

File: docs/config/README.md [hunk 3]
```diff
@@ -405,6 +406,34 @@ style = "green"
...
+| `disabled` | `false`       | Disables the `elm` module.                                |
+
+
+### Example
+
+```toml
+# ~/.config/starship.toml
+
+[elm]
+symbol = "λx.x "
```
Comment (matchai) [inline docs/config/README.md]:
```suggestion
symbol = "λ "
```

**Comment by matchai** [comment]
@all-contributors Please add @m0nhawk for code, docs, and test. 🎉

**Pull request merged**